tests/integration/security/tls_config/main_test.go

```diff
@@ -0,0 +1,57 @@
+//go:build integ
+// +build integ
+
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package tlsconfig
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/tests/integration/security/util"
+)
+
+var (
+	inst istio.Instance
+	apps = &util.EchoDeployments{}
+)
+
+func TestMain(m *testing.M) {
+	framework.
+		NewSuite(m).
+		// Customized setup with TLS configuration
+		Label(label.CustomSetup).
+		Setup(istio.Setup(&inst, setupConfig)).
+		Setup(func(ctx resource.Context) error {
+			return util.SetupApps(ctx, inst, apps, true)
+		}).
+		Run()
+}
+
+func setupConfig(_ resource.Context, cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+	cfg.ControlPlaneValues = `
+values:
+  meshConfig:
+    meshMTLS:
+      minProtocolVersion: TLSV1_3
```

Comment:
I don't think we need a whole new test suite for this feature. A test suite is extremely expensive - about 2 min runtime. If we added one for every little flag in mesh config we would have tests that take a day to complete.

Reply:
The integration test in the PR uses meshConfig to configure the mesh-wide minimum TLS version to be 1.3. When testing a configuration through meshConfig, the convention of all existing integration tests is as follows:
1. In TestMain(), call framework.NewSuite(m). Setup(..., setupConfig)).
2. In setupConfig(), configure the meshConfig.

The integration test in the PR follows this convention. If the meshConfig in the PR is merged into an existing integration test, it changes this existing test since the existing test does not configure minimum TLS version to be 1.3.

Reply:
I would rather have no test than this test as is - adding this much runtime for such a small feature is not viable.

That being said, mesh config can be simply patched at runtime, there are a number of tests that do this: https://github.com/istio/istio/blob/1ade7ec2e569c2302d90a5f6b2bd86872e2314ad/tests/integration/security/normalization_test.go#L227 for example.

Reply:
Based on the comments, the integration test has been removed.